src/main/scala/com/goyeau/kubernetes/client/api/Filterable.scala

```diff
@@ -1,6 +1,6 @@
 package com.goyeau.kubernetes.client.api
 
-trait LabelSelector[A] {
+trait Filterable[A] {
```

Comment:
I'm not a big fan of adding data on the resource objects that will then affect the actions (list, delete...) we do on it.
So I went ahead and pushed a commit on top of your branch to move that at the function level.
Let me know what you think and if I'm missing something.

Reply:
Ok, I am fine with that. There can be more such filters for bulk operations for example fieldSelector. We would need to add more parameters at function level